src/metaswitch/clearwater/plugin_tests/test_default_ifcs_config_plugin.py

```diff
@@ -0,0 +1,110 @@
+
+
+class TestDefaultIFCsConfigPlugin(unittest.TestCase):
+    @mock.patch('clearwater_etcd_plugins.clearwater_config_manager.default_ifcs_config_plugin.safely_write')
+    @mock.patch('clearwater_etcd_plugins.clearwater_config_manager.default_ifcs_config_plugin.run_command')
+    def test_config_changed(self, mock_run_command, mock_safely_write):
+        """Test Config Manager writes new config when config has changed"""
+
+        # Create the plugin
+        plugin = DefaultIFCsConfigPlugin(None)
+
+        # Set up the config strings to be tested
+        old_config_string = "Test config string here. \n More test config string."
+        new_config_string = "This is a different config string. \n Like, totally different."
+
+        # Call 'on_config_changed' with file.open mocked out
+        with mock.patch('clearwater_etcd_plugins.clearwater_config_manager.default_ifcs_config_plugin.open',\
+                        mock.mock_open(read_data=old_config_string), create=True) as mock_open:
+            plugin.on_config_changed(new_config_string, None)
+
+        # Test assertions
+        mock_open.assert_called_once_with(plugin.file(), "r")
+        mock_safely_write.assert_called_once_with(plugin.file(), new_config_string)
+        mock_run_command.assert_called_once_with("/usr/share/clearwater/bin/reload_default_ifcs_config")
+
+
+    @mock.patch('clearwater_etcd_plugins.clearwater_config_manager.default_ifcs_config_plugin.safely_write')
+    @mock.patch('clearwater_etcd_plugins.clearwater_config_manager.default_ifcs_config_plugin.run_command')
+    def test_config_not_changed(self, mock_run_command, mock_safely_write):
+        """Test Config Manager does nothing if called with identical config"""
+
+        # Create the plugin
+        plugin = DefaultIFCsConfigPlugin(None)
+
+        # Set up the config strings to be tested
+        old_config_string = "This is more test config. \n It won't change."
+        new_config_string = old_config_string
+
+        # Call 'on_config_changed' with file.open mocked out
+        with mock.patch('clearwater_etcd_plugins.clearwater_config_manager.default_ifcs_config_plugin.open',\
+                        mock.mock_open(read_data=old_config_string), create=True) as mock_open:
+            plugin.on_config_changed(new_config_string, None)
+
+        # Test assertions
+        mock_open.assert_called_once_with(plugin.file(), "r")
+        mock_safely_write.assert_not_called()
+        mock_run_command.assert_not_called()
+
+    @mock.patch('clearwater_etcd_plugins.clearwater_config_manager.default_ifcs_config_plugin.safely_write')
+    @mock.patch('clearwater_etcd_plugins.clearwater_config_manager.default_ifcs_config_plugin.run_command')
+    def test_default_config_created(self, mock_run_command, mock_safely_write):
+        """Test Config Manager when a new default value is set as etcd key"""
```

Comment:
I feel this test synopsis could be more informative. Maybe something like
"Test Config Manager writes new config when a default value is set as etcd key"
(same for shared config manager test)